File: readability_py/readability.py
from cs50 import get_string

# getting input
text = get_string("Enter the text: ")

letters = 0
sentences = 0
words = 1

# Checking for number of words, sentences and letters on text
for i in range(len(text)):
    if text[i] == " ":
        words += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
    if text[i].isalpha():
        letters += 1

# assign the text parameters to the formula variables
W = 100 / words
L = letters * W
S = sentences * W

# the formula result
index = round(((0.0588 * L) - (0.296 * S)) - 15.8)

# output result
if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: readability_c/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Enter the text: ");
    int letters = 0;
    int sentences = 0;
    int words = 1;
  
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
        else if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
    }
  
    float W = 100.0 / words;
    float L = letters * W;
    float S = sentences * W;
  
    // Here, L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.
    float index = ((0.0588 * L) - (0.296 * S)) - 15.8;
  
    if (round(index) > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }

}